src/pages/Editor.js

```diff
@@ -1,105 +1,411 @@
+		display: 'flex',
+		flexDirection: 'column',
 	},
-	mainEditorContainer: {
-		background: 'rgba(0,0,0,0.9)',
+	leftDrawerComponent: {
 		height: '100vh',
-		width: '60%',
+		background: '#212425',
+		borderRight: '2px solid grey',
+		color: 'white',
+		overflow: 'auto',
+		overflowX: 'hidden',
 	},
-	playVideoSection: {
-		background: 'green',
-		height: '60%',
-		width: '60vw',
+	editorHeader: {
+		position: 'absolute',
 		display: 'flex',
-		flexDirection: 'column',
+		flexDirection: 'row',
 		justifyContent: 'flex-end',
-		backgroundSize: 'contain',
-		backgroundPosition: 'center center',
-		backgroundRepeat: 'no-repeat',
-	},
-	videoActionContainer: {
 		width: '100%',
-		position: 'relative',
-		background: 'transparent',
-		bottom: '0',
-		height: '50px',
+	},
+	videoContainerParent: {
+		background: '#16161e',
+		width: 'inherit',
 		display: 'flex',
-		justifyContent: 'center',
+		flexDirection: 'column',
+		justifyContent: 'space-between',
 		alignItems: 'center',
+		transition: 'all linear 0.5s',
 	},
-	actionBtns: {
-		cursor: 'pointer',
+	videoContainer: {
+		overflow: 'hidden',
+		display: 'flex',
+		flexDirection: 'column',
+		alignItems: 'center',
+		justifyContent: 'center',
+	},
+	bottomDrawerParent: {
+		background: '#16161e',
+		borderTop: '2px solid grey',
+		color: 'white',
+		transition: 'width linear 0.5s',
+		display: 'flex',
+		alignItems: 'center',
 	},
 });
-const Editor = () => {
+
+const Editor = (props) => {
 	const classes = useStyles();
-	const [images, setImages] = useState([
-		'https://picsum.photos/200/300',
-		'https://picsum.photos/200/310',
-		'https://picsum.photos/200/305',
-		'https://picsum.photos/205/300',
-		'https://picsum.photos/210/300',
-		'https://picsum.photos/200/299',
-		'https://picsum.photos/200/301',
-	]);
+	const [isLeftDrawerOpen, setIsLeftDrawerOpen] = React.useState(true);
+	const [isBottomDrawerOpen, setIsBottomDrawerOpen] = React.useState(true);
+	const [transitionLeft, setTransitionLeft] = React.useState(undefined);
+	const [transitionBottom, setTransitionBottom] = React.useState(undefined);
+
+	const handleClickLeft = (Transition) => () => {
+		if (!isLeftDrawerOpen) {
+			setTransitionLeft(() => Transition);
+			setIsLeftDrawerOpen(true);
+		} else {
+			setIsLeftDrawerOpen(false);
+		}
+	};
+	const handleLeftNav = (Transition, curInd) => () => {
+		if (!isLeftDrawerOpen) {
+			setTransitionLeft(() => Transition);
+			setIsLeftDrawerOpen(true);
+		}
+		setActiveIndex(curInd);
+	};
+	const handleClickBottomDrawer = (Transition) => () => {
+		setTransitionBottom(() => Transition);
+		setIsBottomDrawerOpen(true);
+	};
+	const handleCloseBottomDrawer = () => {
+		setIsBottomDrawerOpen(false);
+	};
+	React.useEffect(() => {
+		setTransitionBottom(() => TransitionBottom);
+	}, []);
+
+	// !
+	const [activeIndex, setActiveIndex] = React.useState(0);
+
+	// !VIDEO Video controles functionalities started from here
+	const [isPlaying, setIsPlaying] = React.useState(false);
```

Comment:
You can separate all useStates, useEffects,  all props and useRefs in separate blocks, in whole file. Just from clean code perspective.

Reply:
Sure sir I have separated the code blocks, I'll keep in mind these things next time.